leaderboards/views.py

```diff
@@ -30,16 +30,21 @@ def submit_score(request):
     
 # Create your views here.
 def leaderboard_view_following(request, username):
-    leaderboard_scores = ConnectionsScore.objects.order_by('puzzle_number', 'score_value').reverse()
 
+    print('Switching view')
     #if user is not authenticated, return to leaderboard view with a message saying you must be logged in 
     if not request.user.is_authenticated:
         messages.error(request, "You must be logged in to see following leaderboard.")
-        return redirect(reverse('leaderboards:leaderboard_view'))
-    if username == 'PLACEHOLDER':
+        print('Not Authenticated')
+        return redirect(reverse('leaderboards:leaderboard'))
+    if request.user.username == 'PLACEHOLDER':
         messages.error(request, "You must be logged in to see following leaderboard.")
-        return redirect(reverse('leaderboards:leaderboard_view'))
```

Comment:
Great find! I wasn't exactly sure what was wrong with the reversals but you nailed it.
After looking through `leaderboards/urls.py` we were using the wrong view name. It's defined here
`app_name = 'leaderboards'
urlpatterns = [
    path('', leaderboard_view, name='leaderboard'),`